Applications/WebPortal/WebApp/library/source/pages/account_directory_page/account_directory_page.tsx

```diff
@@ -115,8 +110,8 @@ export class AccountDirectoryPage extends React.Component<Properties, State> {
   }
 
   private onChange(newFilter: string) {
-    this.setState({filter: newFilter});
     this.setState({
+      filter: newFilter,
       openedGroups: new Beam.Set<Beam.DirectoryEntry>()
     });
   }
```

Comment:
Still need to handle closing the card.

Reply:
Won't the change in state cause the cards to close? It's only open if it's in the open set, but when the filter changes I replace the current set with the empty set.

Reply:
Looks like I didn't save the changes.

Reply:
The comment about closing cards is not specific to any line of code, it's just a general feature that needs to get implemented as part of this task.